hummingbird/ml/_topology.py

```diff
@@ -185,6 +218,79 @@ def fix_graph(graph):
             return num_fixed
 
         fix_graph(hb_model.graph)
+    elif backend == tvm_backend:
+        # First we need to generate the torchscript model.
+        batch_trace_input, remainder_trace_input = _get_trace_input_from_test_input(
+            extra_config[constants.TEST_INPUT], batch_size
+        )
+        ts_model = _jit_model(torch_model, batch_trace_input, "cpu", extra_config)
+        if remainder_trace_input is not None:
+            remainder_ts_model = _jit_model(torch_model, remainder_trace_input, "cpu", extra_config)
+
+        # Generate the test input in the TVM format. In case we have a remainder beyond the batch, generate a remainder test input as well.
+        test_input = [
+            (
+                topology.raw_model.input_names[i],
+                batch_trace_input[i].shape if type(batch_trace_input) is tuple else batch_trace_input.shape,
+            )
+            for i in range(len(topology.raw_model.input_names))
+        ]
+        if remainder_trace_input is not None:
+            remainder_test_input = [
+                (
+                    topology.raw_model.input_names[i],
+                    remainder_trace_input[i].shape if type(remainder_trace_input) is tuple else remainder_trace_input.shape,
+                )
+                for i in range(len(topology.raw_model.input_names))
+            ]
+
+        # Pick the proper target.
+        if device == "cuda":
+            target = tvm.target.cuda()
+            ctx = tvm.gpu()
+        elif device == "cpu":
+            target = "llvm"
+            ctx = tvm.cpu()
+        elif "llvm" in device:
+            target = device
+            ctx = tvm.cpu()
+        else:
+            raise RuntimeError("Device {} not recognized".format(device))
+
+        # Get configuration parameters.
+        config = {}
+        if constants.TVM_MAX_FUSE_DEPTH in extra_config:
+            config["relay.FuseOps.max_depth"] = extra_config[constants.TVM_MAX_FUSE_DEPTH]
+        else:
+            # 50 is a good depth for operator fusion. More than that will probably hurt performance.
+            # https://github.com/microsoft/hummingbird/issues/232#issuecomment-697979508
+            config["relay.FuseOps.max_depth"] = 50
+
+        # Create the relay version of the model.
+        model, params = relay.frontend.from_pytorch(ts_model, test_input)
+        if remainder_trace_input is not None:
+            remainder_model, remainder_params = relay.frontend.from_pytorch(remainder_ts_model, remainder_test_input)
+
+        # Generate the model.
+        with tvm.transform.PassContext(opt_level=3, config=config):
+            opt_mod, opt_params = relay.optimize(model, target=target, params=params)
```

Comment:
no need to call optimize explicitly, `relay.build` calls `optimize` inside

Reply:
Ah, good to know! Let me fix it quickly.